src/test/steps/createCampaignSteps.ts

```diff
@@ -0,0 +1,12 @@
+import { When } from '@cucumber/cucumber';
+import basePage from '../../pages/basePage';
+
+When('User selects {string} company for {string} on {string}', async function (company, type, page) {
+    const [companyName, companyType, pageName] = basePage.wrapper.toCamelCase(company, type, page);
+    await basePage.wrapper.waitAndClick(basePage[pageName][companyType]) 
+
+    await basePage.wrapper.waitAndClick(basePage[pageName].companyNameInput) 
+    await basePage.wrapper.type(basePage[pageName].companyNameInput, companyName);
+    await basePage.wrapper.pressKeyboard('Enter');
+    await basePage.wrapper.waitAndClick(basePage[pageName][companyName+'Company']);
```

Comment:
Not sure why you always need to append `+'Company'`!?

Reply:
I am selecting the advertiser company that I use for creating a campaign. I am receiving only its name and in the variable, there is word 'company' after it.

Reply:
Why don't you pass full company name in the scenario step?